src/steps/os/unix.rs

```diff
@@ -343,11 +344,28 @@ pub fn run_sdkman(base_dirs: &BaseDirs, cleanup: bool, run_type: RunType) -> Res
 
     print_separator("SDKMAN!");
 
-    let cmd_selfupdate = format!("source {} && sdk selfupdate", &sdkman_init_path);
-    run_type
-        .execute(&bash)
-        .args(&["-c", cmd_selfupdate.as_str()])
-        .check_run()?;
+    let sdkman_config_path = env::var("SDKMAN_DIR")
+        .map(PathBuf::from)
+        .unwrap_or_else(|_| base_dirs.home_dir().join(".sdkman"))
+        .join("etc")
+        .join("config")
+        .require()
+        .map(|p| format!("{}", &p.display()))?;
```

Comment:
You can probably remove this mapping since [`Ini::load_from_file`](https://docs.rs/rust-ini/latest/ini/struct.Ini.html#method.load_from_file) takes `AsRef<Path>`